src/nautilus_librarian/mods/libvips/domain/process_image.py

```diff
@@ -0,0 +1,25 @@
+import os
+
+from nautilus_librarian.mods.libvips.domain.utils import (
+    get_image,
+    modify_icc_profile,
+    resample_image,
+    save_image,
+)
+
+
+def create_output_folder(destination_filename):
+    os.makedirs(os.path.dirname(destination_filename), exist_ok=True)
+
+
+def process_image(
```

Comment:
rename to `generate_base_image`?

Reply:
Well, actually, I think that although we are in the domain layer, as the action accepts as parameter the ICC profile and the image size, this processing could be used also for any resample and/or ICC modification (i.e. base to web-ready, base to book-ready, even acquired to gold...), so I preferred to use a more generic naming.
Maybe a better name could be "generate_resampled_reprofiled_image"